classes/Parameters/ConfigurationValidator.php

```diff
@@ -107,9 +107,12 @@ private function validateArchiveXml(string $xml, bool $isLocal): ?string
         return null;
     }
 
-    private function validateBool(string $boolValue, string $key): ?string
+    /**
+     * @param string|bool $boolValue
+     */
+    private function validateBool($boolValue, string $key): ?string
     {
-        if ($boolValue === '' || filter_var($boolValue, FILTER_VALIDATE_BOOLEAN, FILTER_NULL_ON_FAILURE) === null) {
+        if (!is_bool($boolValue) && ($boolValue === '' || filter_var($boolValue, FILTER_VALIDATE_BOOLEAN, FILTER_NULL_ON_FAILURE) === null)) {
             return $this->translator->trans('Value must be a boolean for %s', [$key]);
         }
```

Comment:
Let's remember these methods return an error when something is wrong. If we go with another method, this new one would return a boolean to let us know if further checks are needed or not. This makes the contents of the class more confusing.

Having this in the switch / case list adds unnecessary complexity as well. We have calls to a single method for each case and I like the way it currently is. Simple and efficient.

The `mixed` case does not apply here as we specify two types in the php doc block. Even if the freshly introduce lvl 10 of PHPStan, this definition pass:
![image](https://github.com/user-attachments/assets/e8c37ed9-f562-4c32-a7bd-ca70bbae5cfd)

Reply:
> Having this in the switch / case list adds unnecessary complexity as well.

I don't see why this would add complexity, we would just have the check before calling the method, which could be renamed to explain that we are validating a boolean "string"

PHPstan may not say anything, but this remains a practice that does not follow modern PHP practices. This would not have been possible on languages ​​with (really) strong typing, and I think it's better.

I think that on the contrary, a parameter with multiple types can make the function more difficult to understand, because it can be difficult to know how the function will behave depending on the type passed.

Reply:
But I don't want to block the PR for this detail, we have to move it forward :+1: